docs/src/main/sphinx/security/authentication-types.rst

```diff
@@ -0,0 +1,60 @@
+====================
+Authentication types
+====================
+
+Trino supports multiple authentication types to ensure all users of the system
+are authenticated. Different authenticators allow user management in one or more
+systems. Using :doc:`TLS <tls>` is required for all authentications types.
+
+You can configure one or more authentication types with the
+``http-server.authentication.type`` property. The following authentication types
+and authenticators are available:
+
+* ``PASSSWORD`` for  :doc:`password-file`, :doc:`ldap`, and :doc:`salesforce`
+* ``OAUTH2`` for :doc:`oauth2`
+* ``CERTIFICATE`` for certificate authentication
+* ``JWT`` for Java Web Token (JWT) authentication
+* ``KERBEROS`` for :doc:`Kerberos authentication <server>`
+
+The simplest configuration to get started is password authentication backed by a
+password file:
+
+.. code-block:: properties
+
+    http-server.authentication.type=PASSWORD
+
+
+Multiple authentication types
+-----------------------------
+
+You can use multiple authentication types, separating with commas in the
+configuration:
+
+.. code-block:: properties
+
+    http-server.authentication.type=PASSWORD,CERTIFICATE
+
+Authentication is performed in order of the entries, and first successful
+authentication results in access.
+
+Multiple password authenticators
+--------------------------------
+
+You can use multiple password authenticators types by referencing multiple
+configuration files:
+
+.. code-block:: properties
+
+    http-server.authentication.type=PASSWORD,CERTIFICATE
+    password-authenticator.config-files=etc/ldap1.properties,etc/ldap2.properties,etc/password.properties
+
+In the preceding example, the configuration files ``ldap1.properties`` and
+``ldap1.properties`` are regular :doc:`LDAP authenticator configuration files
+<ldap>`. The ``password.properties`` is a :doc:`password file authenticator
+configuration file <password-file>`.
+
+Any user authentication credentials first contact the LDAP server from
```

Comment:
- Remove "Any"
- credentials don't contact things. Maybe, "credentials are first examined by"

Reply:
Maybe something like

> The credentials are first validated against the LDAP server from `ldap1`, then against the LDAP server from `ldap2`, then finally against the password file. The first successful authentication results in access, and none of the subsequent authenticators are called.

I changed "accessed" to "called" in the second sentence, since that's more reflective of what happens (and a reason someone might care is if they have a custom authenticator that has a side effect like logging).

Reply:
Updated